modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java

```diff
@@ -1253,6 +1260,60 @@ public void testUpdateDisplayDateExpirationDateReviewDate()
 		Assert.assertEquals(reviewDate, dlFileEntry.getReviewDate());
 	}
 
+	@Test
+	public void testUpdateFileEntryDynamicQueryTransactionInterval()
+		throws Exception {
+
+		List<DLFileEntry> dlFileEntries = new ArrayList<>();
+
+		for (int i = 0; i < 20; i++) {
+			dlFileEntries.add(
+				DLFileEntryLocalServiceUtil.addFileEntry(
+					null, TestPropsValues.getUserId(), _group.getGroupId(),
+					_group.getGroupId(),
+					DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+					StringUtil.randomString(), ContentTypes.TEXT_PLAIN,
+					StringUtil.randomString(), StringUtil.randomString(),
+					StringPool.BLANK, StringPool.BLANK, -1, new HashMap<>(),
+					null, new ByteArrayInputStream(new byte[0]), 0, null, null,
+					null,
+					ServiceContextTestUtil.getServiceContext(
+						_group.getGroupId(), TestPropsValues.getUserId())));
+		}
+
+		AtomicInteger atomicInteger = new AtomicInteger(0);
+
+		try {
+			DLFileEntryLocalServiceUtil.forEachFileEntry(
+				TestPropsValues.getCompanyId(),
+				dlFileEntry -> {
+					if (atomicInteger.addAndGet(1) < 20) {
+						dlFileEntry.setDescription("Updated");
+					}
+					else {
+						throw new RuntimeException();
+					}
+				},
+				0L, new String[] {ContentTypes.TEXT_PLAIN});
+		}
+		catch (RuntimeException runtimeException) {
```

Comment:
@manuelRives, doesn't the SF complains about this unprocessed exception?

Reply:
Surprisingly it was not complaining, but I agree that it is better to process it. Fixed.